core/src/main/java/com/linecorp/armeria/server/AnnotatedHttpServiceConfigurator.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+
+import com.linecorp.armeria.internal.annotation.AnnotatedHttpService;
+import com.linecorp.armeria.server.annotation.ExceptionHandlerFunction;
+import com.linecorp.armeria.server.annotation.RequestConverterFunction;
+import com.linecorp.armeria.server.annotation.ResponseConverterFunction;
+
+/**
+ * A configurator to configure {@link ExceptionHandlerFunction}s, {@link RequestConverterFunction}s
+ * or {@link ResponseConverterFunction}s for an {@link AnnotatedHttpService}.
+ */
+final class AnnotatedHttpServiceConfigurator {
+
+    /**
+     * Creates a new instance with the specified {@code exceptionHandlersAndConverters}.
+     */
+    static AnnotatedHttpServiceConfigurator ofExceptionHandlersAndConverters(
+            Iterable<?> exceptionHandlersAndConverters) {
+
+        Builder<ExceptionHandlerFunction> exceptionHandlers = null;
```

Comment:
I think we can just assign `ImmutableList.builder()` here and remove all `if` conditions to make this simple